website/static/js/pages/project-dashboard-page.js

```diff
@@ -213,22 +238,26 @@ $(document).ready(function () {
         }
     });
 
-    //Clear input fields on Add Component Modal
-    $('#confirm').on('click', function () {
-        $('#alert').text('');
-        $('#title').val('');
-        $('#category').val('');
-    });
-
-    // only focus input field on modals when not IE
-    $('#newComponent').on('shown.bs.modal', function(){
-        if(!$osf.isIE()){
-            $('#title').focus();
-        }
-    });
+    loadCategories().then(function(categoryList) {
+        var addProjectTemplate = m.component(AddProject, {
+            buttonTemplate: m('.btn.btn-sm.btn-default[data-toggle="modal"][data-target="#addSubComponent"]', {onclick: function() {
+                $osf.trackClick('project-dashboard', 'add-component', 'open-add-project-modal');
+            }}, 'Add Component'),
+            modalID: 'addSubComponent',
+            title: 'Create new component',
+            parentID: window.contextVars.node.id,
+            parentTitle: window.contextVars.node.title,
+            categoryList: categoryList,
+            stayCallback: function () {
+                var ap = this; // AddProject controller
+            },
+            trackingCategory: 'project-dashboard',
+            trackingAction: 'add-component',
+            contributors: window.contextVars.node.contributors,
+            currentUserCanEdit: window.contextVars.currentUser.canEdit
+        });
 
-    $('#newComponent').on('hidden.bs.modal', function(){
-        $('#newComponent .modal-alert').text('');
+        m.mount(document.getElementById('newComponent'), m.component(addProjectTemplate, {wrapperSelector : '#addSubComponent'}));
```

Comment:
Rendering the button after we load the categories results in the button being flashed on screen:

![button](https://cloud.githubusercontent.com/assets/2379650/16421580/811f7812-3d23-11e6-9370-dc66860885bf.gif)

This flash shouldn't be necessary. The user has to go through a few clicks to get to the Category dropdown, which should provide enough time for the OPTIONS request to finish.

I suggest we either

1. Render the button immediately, then send the OPTIONS request on page load, or
2. Render the button immediately, then send the OPTIONS request when the user clicks the "Add component" button.

This isn't necessarily a blocker to getting this merged, but it's worth looking into. @leb2dg Can you take a look?